src/main/scala/dsptools/numbers/chisel_concrete/DspComplex.scala

```diff
@@ -0,0 +1,82 @@
+// See LICENSE for license details.
+
+package dsptools.numbers
+
+import chisel3._
+import chisel3.experimental.FixedPoint
+import dsptools.DspException
+import implicits._
+import breeze.math.Complex
+
+object DspComplex {
+
+  def apply[T <: Data:Ring](gen: T): DspComplex[T] = {
+    if (gen.isLit()) throw DspException("Cannot use Lit in single argument DspComplex.apply")
+    apply(gen, gen)
+  }
+
+  // If real, imag are literals, the literals are carried through
+  // In reality, real and imag should have the same type, so should be using single argument
+  // apply if you aren't trying t create a Lit
+  def apply[T <: Data:Ring](real: T, imag: T): DspComplex[T] = {
+    val newReal = if (real.isLit()) real else real.cloneType
+    val newImag = if (imag.isLit()) imag else imag.cloneType
+    new DspComplex(newReal, newImag)
+  }
+
+  // Needed for assigning to results of operations; should not use in user code for making wires
+  // Assumes real, imag are not literals
+  private [dsptools] def wire[T <: Data:Ring](real: T, imag: T): DspComplex[T] = {
+    val result = Wire(DspComplex(real.cloneType, imag.cloneType))
+    result.real := real
+    result.imag := imag
+    result
+  }
+
+  // Constant j
+  def j[T <: Data:Ring] : DspComplex[T] = DspComplex(Ring[T].zero, Ring[T].one)
+
+  // Creates a DspComplex literal of type DspComplex[T] from a Breeze Complex
+  // Note: when T is FixedPoint, the # of fractional bits is determined via DspContext
+  def apply[T <: Data:Ring:ConvertableTo](c: Complex): DspComplex[T] = {
+    DspComplex(ConvertableTo[T].fromDouble(c.real), ConvertableTo[T].fromDouble(c.imag))
+  }
+  // Creates a DspComplex literal where real and imaginary parts have type T (and binary point 
+  // determined by binaryPoint of t)
+  def proto[T <: Data:Ring:ConvertableTo](c: Complex, t: T): DspComplex[T] = {
+    DspComplex(ConvertableTo[T].fromDouble(c.real, t), ConvertableTo[T].fromDouble(c.imag, t))
+  }
+  // Creates a DspComplex literal where real and imaginary parts have type T (width/binary point 
+  // determined by width/binaryPoint of t)
+  def protoWithFixedWidth[T <: Data:Ring:ConvertableTo](c: Complex, t: T): DspComplex[T] = {
+    DspComplex(ConvertableTo[T].fromDoubleWithFixedWidth(c.real, t), 
+      ConvertableTo[T].fromDoubleWithFixedWidth(c.imag, t))
+  }
+
+}
+
+class DspComplex[T <: Data:Ring](val real: T, val imag: T) extends Bundle {
+  // Multiply by j
+  def j: DspComplex[T] = DspComplex.wire(-imag, real)
```

Comment:
I think it is confusing to call this `j`, especially because the companion object also has a function called `j` which has different behavior. I'm not sure what to call it. `multiplyByJ` is verbose but impossible to misunderstand. Something like `x_j` or `multj` that is shorter might be OK too.